pkg/cloud/openstack/clients/secgroupservice.go

```diff
@@ -0,0 +1,348 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"fmt"
+
+	"k8s.io/klog"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/groups"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/rules"
+
+	openstackconfigv1 "sigs.k8s.io/cluster-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
+)
+
+const (
+	secGroupPrefix     string = "k8s"
+	controlPlaneSuffix string = "controlplane"
+	globalSuffix       string = "all"
+)
+
+var defaultRules = []openstackconfigv1.SecurityGroupRule{
```

Comment:
nit: It would be nice to add a comment here explaining what the rules below do. Who/where they provide access for/to

Reply:
Good point, I kinda assumed that anyone looking at this would know about OpenStack secgroups but that might be a bit much to ask for.